Backgrounds/AstralDesertSurfaceBGStyle.cs

```diff
@@ -1,37 +1,147 @@
-﻿using Terraria.ModLoader;
+﻿using Microsoft.Xna.Framework.Graphics;
+using System.Reflection;
+using System;
+using Terraria.Graphics.Effects;
+using Terraria;
+using Terraria.ModLoader;
+using Microsoft.Xna.Framework;
 
 namespace CalamityMod.Backgrounds
 {
     public class AstralDesertSurfaceBGStyle : ModSurfaceBackgroundStyle
     {
-        public override int ChooseFarTexture() => BackgroundTextureLoader.GetBackgroundSlot("CalamityMod/Backgrounds/BlankPixel");
-
-        public override int ChooseMiddleTexture() => BackgroundTextureLoader.GetBackgroundSlot("CalamityMod/Backgrounds/BlankPixel");
-
-        public override int ChooseCloseTexture(ref float scale, ref double parallax, ref float a, ref float b)
-        {
-            b -= 75f;
-            return BackgroundTextureLoader.GetBackgroundSlot("CalamityMod/Backgrounds/BlankPixel");
-        }
+        readonly int CloseBGYOffset = 475;
+        readonly int MiddleBGYOffset = 475;
 
         public override void ModifyFarFades(float[] fades, float transitionSpeed)
         {
-            // This just fades in the background and fades out other backgrounds.
             for (int i = 0; i < fades.Length; i++)
             {
                 if (i == Slot)
                 {
                     fades[i] += transitionSpeed;
                     if (fades[i] > 1f)
+                    {
                         fades[i] = 1f;
+                    }
                 }
                 else
                 {
                     fades[i] -= transitionSpeed;
                     if (fades[i] < 0f)
+                    {
                         fades[i] = 0f;
+                    }
+                }
+            }
+        }
+
+        public override int ChooseFarTexture()
+        {
+            return BackgroundTextureLoader.GetBackgroundSlot("CalamityMod/Backgrounds/AstralSurfaceHorizon");
+        }
+
+        public override int ChooseMiddleTexture()
+        {
+            return BackgroundTextureLoader.GetBackgroundSlot("CalamityMod/Backgrounds/AstralDesertSurfaceFar");
+        }
+
+        public override int ChooseCloseTexture(ref float scale, ref double parallax, ref float a, ref float b)
+        {
+            return BackgroundTextureLoader.GetBackgroundSlot("CalamityMod/Skies/AstralDesertSurfaceMiddle");
+        }
+
+        public override bool PreDrawCloseBackground(SpriteBatch spriteBatch)
+        {
+            float screenOff = (float)typeof(Main).GetField("screenOff", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(Main.instance);
+            float scAdj = (float)typeof(Main).GetField("scAdj", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(Main.instance);
+            Color COSBMAplha = (Color)typeof(Main).GetField("ColorOfSurfaceBackgroundsModified", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null);
+            Color ColorOfSurfaceBackgroundsModified = new Color(63, 51, 90, COSBMAplha.A);
```

Comment:
Background drawing is already terribly inefficient; cache your reflection.

Reply:
I would personally love to use access modifiers but we don't provide a standard `.csproj`  yet. TODO for myself to bring up, maybe?